includes/fbbot/class-omise-fbbot-endpoints.php

```diff
@@ -0,0 +1,168 @@
+				array(
+					'methods' => WP_REST_Server::CREATABLE,
+					'callback' => array( $this, 'callback_omise_webhook' )
+				)
+			) 
+		);
+
+		register_rest_route( $namespace, '/callback_fbbot_checkout', array(
+				array(
+					'methods' => WP_REST_Server::CREATABLE,
+					'callback' => array( $this, 'callback_fbbot_checkout' )
+				)
+			) 
+		);
+	
+	}
+
+	public function verify_fb_token_callback( $request ) {
+		$params = $request->get_query_params();
+		echo $params['hub_challenge'];
+		die();
+  	}
+
+  	public function verify_fb_token_request( $request = NULL ) {
+		$params = $request->get_query_params();
+
+		if ( $params && isset( $params['hub_challenge'] ) && $params['hub_verify_token'] == $this->facebook_page_verify_token ) {
+			return true;
+		}
+
+		return false;
+	}
+
+  	public function callback_fb_webhook ( $request ) {
+		$params = $request->get_params();
+
+		if ( ! ( $params && $params['entry'] ) ) {
+		  	return;
+		}
+
+		foreach ( (array) $params['entry'] as $entry ) {
+			if ( ! ( $entry && $entry['messaging'] ) ) {
+				break;
+			}
+
+			foreach ( (array) $entry['messaging'] as $messaging_event ) {
+				if ( isset( $messaging_event['message'] ) ) {
+					if ( isset( $messaging_event['message']['is_echo'] ) ) {
+						break;
+					}
+
+					if ( isset( $messaging_event['message']['quick_reply'] ) ) {
+						break;
+					}
+
+					$sender_id = $messaging_event['sender']['id'];
+
+					// Handle text message
+					$text = $messaging_event['message']['text'];
+					Omise_FBBot_Request_Handler::handle_message_from( $sender_id, $text );
+					break;
+
+				} else if ( isset( $messaging_event['postback'] ) ) {
+					// Handle payload
+					$sender_id = $messaging_event['sender']['id'];
+					$payload = $messaging_event['postback']['payload'];
+
+					Omise_FBBot_Request_Handler::handle_payload_from( $sender_id, $payload );
+				} else {
+				  	// Unused case
+				  	break;
+				}
+			} // foreach ['messaging']
+
+		} // foreach ['entry']
+	}
+
+	public function callback_omise_webhook( $request ) {
+		Omise_FBBot_Request_Handler::handle_callback_omise_webhook( $request );
+	}
+
+  	public function callback_fbbot_checkout( $request ) {
+		$params = $request->get_params();
+
+	    try {
+	        if ( ! $user = Omise_FBBot_User_Service::get_user( $params['messenger_id'] ) ) {
+	            throw new Exception( __( "Oop! We got some problem can't get your profile. Please try again later.", 'omise' ) );
+	        }
+
+	        $params['address'] = array(
+	            'first_name' => $user['first_name'],
+	            'last_name'  => $user['last_name'],
+            	'email'      => $params['customer_email']
+	        );
+
+	        if ( ! $order = Omise_FBBot_WooCommerce::create_order( $params ) ) {
+	            throw new Exception( __( "Oop! We can't create your order.", 'omise' ) );
+	        }
+
+	        $params['metadata'] = array(
+				'source' => 'woo_omise_bot',
+				'product_id' => $params['product_id'],
```

Comment:
@igroomgrim since we attach a product object into order object already. We better to use `product_id` from object instead. To make sure that these value will be the same.